distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/APMJvmOptionsTests.java

```diff
@@ -74,46 +75,61 @@ public void testFileDeleteWorks() throws IOException {
     }
 
     public void testExtractSettings() throws UserException {
-        Settings settings = Settings.builder()
-            .put("tracing.apm.enabled", true)
-            .put("tracing.apm.agent.server_url", "https://myurl:443")
-            .put("tracing.apm.agent.service_node_name", "instance-0000000001")
-            .put("tracing.apm.agent.global_labels.deployment_id", "123")
-            .put("tracing.apm.agent.global_labels.deployment_name", "APM Tracing")
-            .put("tracing.apm.agent.global_labels.organization_id", "456")
-            .build();
-
-        var extracted = APMJvmOptions.extractApmSettings(settings);
-
-        assertThat(
-            extracted,
-            allOf(
-                hasEntry("server_url", "https://myurl:443"),
-                hasEntry("service_node_name", "instance-0000000001"),
-                hasEntry(equalTo("global_labels"), not(endsWith(","))), // test that we have collapsed all global labels into one
-                not(hasKey("global_labels.organization_id")) // tests that we strip out the top level label keys
+        Settings settings;
+        for (String prefix : List.of("tracing.apm.agent.", "telemetry.agent.")) {
+            settings = Settings.builder()
+                .put("tracing.apm.enabled", true)
+                .put(prefix + "server_url", "https://myurl:443")
+                .put(prefix + "service_node_name", "instance-0000000001")
+                .put(prefix + "global_labels.deployment_id", "123")
+                .put(prefix + "global_labels.deployment_name", "APM Tracing")
+                .put(prefix + "global_labels.organization_id", "456")
+                .build();
+
+            var extracted = APMJvmOptions.extractApmSettings(settings);
+
+            assertThat(
+                extracted,
+                allOf(
+                    hasEntry("server_url", "https://myurl:443"),
+                    hasEntry("service_node_name", "instance-0000000001"),
+                    hasEntry(equalTo("global_labels"), not(endsWith(","))), // test that we have collapsed all global labels into one
+                    not(hasKey("global_labels.organization_id")) // tests that we strip out the top level label keys
+                )
+            );
+
+            List<String> labels = Arrays.stream(extracted.get("global_labels").split(",")).toList();
+
+            assertThat(labels, hasSize(3));
+            assertThat(labels, containsInAnyOrder("deployment_name=APM Tracing", "organization_id=456", "deployment_id=123"));
+
+            settings = Settings.builder()
+                .put("tracing.apm.enabled", true)
+                .put(prefix + "server_url", "https://myurl:443")
+                .put(prefix + "service_node_name", "instance-0000000001")
+                .put(prefix + "global_labels.deployment_id", "")
```

Comment:
How about collapsing these cases into above settings? I was briefly confused why repeating the same thing. It could also be worth adding a short comment for each. Consider it optional, this hasn't changed.